localstack/localstack
# fix: replace `DEVELOP` mode

## Motivation

During the migration to the new runtime ([PR 1 adding the new runtime](https://github.com/localstack/localstack/pull/10942), [PR 2 removing the old runtime](https://github.com/localstack/localstack/pull/11202)), the `DEVELOP` live debugging functionality was missed. This workflow, coupled with using `localstack.dev.run` is crucial for debugging parts of LocalStack that _only_ work when run from a container, e.g. testing inter-container Docker networking via services.



## Changes

- Add the previous `DEVELOP` functionality back, including launching the `debugpy` debugger
  - enable the feature with `DEVELOP=1`
  - pause the LocalStack start on debugger connection with `WAIT_FOR_DEBUGGER=1`
  - configure the port used by `debugpy` with `DEVELOP_PORT` (default: 5678)
- Fix a bug in `localstack.dev.run` where community only entrypoints were not being mounted from the correct place (when launched from a community checkout)
- Fix a bug in `localstack.dev.run` where community and pro entrypoints were not being mounted from the correct place (when launched from a pro checkout)
- Add convenience properties on `HostPaths` to access source code checkout paths on the host

## Testing

* checkout `master`
* run `localstack.dev.run --mount-entrypoints`
    * notice that the community entrypoints file is not mounted into the container
* checkout this PR
* run `localstack.dev.run --mount-entrypoints`
    * notice that the community entrypoints file **is** mounted into the container
* run `localstack.dev.run --mount-entrypoints --develop`
    * notice that the logs indicate that `debugpy` is being installed, and is available on port 5678
